core/block_render_svg_vertical.js

```diff
@@ -857,6 +940,55 @@ Blockly.BlockSvg.prototype.renderDrawRight_ = function(steps,
 };
 
 /**
+ * Render the shape for an input.
+ * If there's a connected block, hide the input shape.
+ * Otherwise, draw and set the position of the input shape.
+ * @param {!Blockly.Input} input Input to be rendered.
+ * @param {Number} x X offset of input.
+ * @param {Number} y Y offset of input.
+ * @return {Number} Amount cursor should be offset (depending on width of render)
+ */
+Blockly.BlockSvg.prototype.renderInputShape_ = function(input, x, y) {
+  var inputShape = this.inputShapes_[input.name];
+  var inputShapeWidth = 0;
+  // Input shapes are only visibly rendered on non-connected non-insertion-markers.
+  if (!input.connection.targetConnection && !this.isInsertionMarker()) {
```

Comment:
Flip order of if/else to
if (this.isInsertionMarker() || input.connection.targetConnection) { hide} else { calculate + show }

Optionally, use connection.isConnected() instead of connection.targetConnection.  The latter is public but I wish it weren't.